waku/v2/node/peer_manager/peer_manager.nim

```diff
@@ -455,10 +455,41 @@ proc getNumStreams*(pm: PeerManager, dir: Direction, protocol: string): int =
             numConns += 1
   return numConns
 
+proc getNumConnsAndStreams*(pm: PeerManager, protocol: string): (int, int, int, int) =
```

Comment:
Well, I don't have a strong preference, my goal here was to come up with the most optimal code possible (which I am pretty sure my code is not anyway:D). So if you prefer to have it separated, I can update the PR

Reply:
since imho this optimization is not very critical (runs every few seconds on a few dozen of peers), i would say its better to sacrifice some iterations in favor of better readability and have a more generic function. eg someone may want to know the connections but not the streams. Having everything in one place seems to specific for this use case. That's why i think it's better to keep it separate.

So I would have 2 functions but return both directions as in your idea.

Reply:
See the latest push, I also updated the PR description and removed WIP flag